### ISSUE TITLE
bug: Moq1208 does not detect Returns<T>(delegate) with explicit generic argument on async methods

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#1091]
## Summary

The Moq1208 analyzer (ReturnsDelegateShouldReturnTaskAnalyzer) fails to detect `Returns<T>(delegate)` calls with explicit generic type arguments on async methods. This is a false negative.

## Reproduction

```csharp
var mock = new Mock<IFoo>();

// Moq1208 should fire here but does NOT
// AsyncMethod returns Task<int>, but the delegate returns int (not Task<int>)
mock.Setup(m => m.AsyncMethod())
    .Returns<string>(s => s.Length);
```

Compare with the non-generic form, which IS correctly detected:

```csharp
// Moq1208 correctly fires here
mock.Setup(m => m.AsyncMethod())
    .Returns(() => 42);
```

## Root Cause

Moq has 17 `Returns()` overloads including `Returns<T1>(Func<T1, TResult>)` through `Returns<T1..T16>(...)`. These are method-level generics parameterizing the delegate argument types.

When a user writes `.Returns<string>(s => s.Length)` on a `Task<int>` method, Roslyn's target-type inference resolves the lambda's return type from the target delegate signature (`Func<string, Task<int>>`), not from the lambda body. The analyzer's `GetDelegateReturnType` then sees `Task<int>` (inferred by the compiler), finds no mismatch with the method's return type, and suppresses the diagnostic.

The lambda body actually returns `int` (from `s.Length`), which is the real type mismatch. The compiler wraps this in an implicit conversion to `Task<int>` because of the target-type context.

## Impact

Users writing `Returns<T>(delegate)` on async methods get no warning. At runtime, Moq throws:
```
Invalid callback. Setup on method with return type 'Task<int>' cannot invoke callback with return type 'int'.
```

## Suggested Fix

The analyzer should inspect the lambda body's return type directly rather than relying solely on `GetSymbolInfo` for the delegate return type. When an explicit generic argument is present (GenericNameSyntax), the analyzer should:

1. Extract the lambda expression from the argument
2. Analyze the lambda body's actual return type
3. Compare against the async method's unwrapped return type (T from Task<T>)

## Note

The code fix (`ReturnsDelegateShouldReturnTaskFixer.cs`) already handles `GenericNameSyntax` correctly. It preserves the `<T>` type argument when rewriting `Returns<T>` to `ReturnsAsync<T>`. The fixer is ready; only the analyzer detection needs fixing.

## Related

- PR #1089: Added documentation tests for this gap
- Issue #777: Parent tracking issue for Moq1208 completion
- Moq source: `IReturns<TMock, TResult>` defines `Returns<T1>` through `Returns<T1..T16>` overloads

**Comment by rjmurillo** [comment]
Correction after investigation: the title says "Moq1208" but the analyzer file is `ReturnsDelegateShouldReturnTaskAnalyzer.cs`. Both confirmed present. Root cause matches the traycer.ai plan: `GetDelegateReturnType` trusts `GetSymbolInfo` on the lambda, which target-type inference resolves to `Task<int>`, so no mismatch fires for `.Returns<string>(s => s.Length)`. Fix: when `Returns` uses `GenericNameSyntax` and the arg is a lambda, inspect the body directly (`GetTypeInfo` / `GetReturnTypeFromBlock`) instead. Confirmed false-negative, scope verified against current src.
